linera-witty/src/runtime/memory.rs

```diff
@@ -11,32 +11,36 @@ use crate::{Layout, WitType};
 use frunk::{hlist, hlist_pat, HList};
 use std::borrow::Cow;
 
+#[cfg(test)]
+#[path = "unit_tests/memory.rs"]
+mod tests;
+
 /// An address for a location in a guest WebAssembly module's memory.
 #[derive(Clone, Copy, Debug, Eq, PartialEq)]
 pub struct GuestPointer(pub(crate) u32);
 
 impl GuestPointer {
     /// Returns a new address that's the current address advanced to add padding to ensure it's
     /// aligned to the `alignment` byte boundary.
-    pub fn aligned_at(&self, alignment: u32) -> Self {
+    pub const fn aligned_at(&self, alignment: u32) -> Self {
         let padding = (-(self.0 as i32) & (alignment as i32 - 1)) as u32;
```

Comment:
Maybe comment something like
```
// Compute `-self.0` modulo `2^alignment`
```

Reply:
Done, and linked to Wikipedia where I first found the formula :laughing: